utils/appliance/__init__.py

```diff
@@ -483,6 +497,21 @@ def ssh_client(self):
             'container': self.container,
         }
         ssh_client = ssh.SSHClient(**connect_kwargs)
+        try:
+            ssh_client.get_transport().is_active()
+            logger.error('default appliance ssh credentials are valid')
+        except Exception as e:
+            logger.error(e)
+            logger.error('default appliance ssh credentials failed, trying establish ssh connection'
+                         ' using ssh private key')
+            with open(conf_path.join('appliance_private_key').strpath, 'w') as key:
+                key.write(conf.credentials['ssh']['private_key'])
+            connect_kwargs = {
+                'hostname': self.hostname,
+                'username': conf.credentials['ssh']['ssh-user'],
+                'key_filename': conf_path.join('appliance_private_key').strpath,
+            }
+            ssh_client = ssh.SSHClient(**connect_kwargs)
```

Comment:
Consider writing a small function for this since its fairly complex for an except clause. There are chances for more exceptions to occur here.

You could also make the log message at the start of this clause a warning, since you're trying a second connection. If the second fails, then elevate to 'error' level.